pkg/controllers/federatedcluster/clusterjoin.go

```diff
@@ -287,17 +288,18 @@ func createAuthorizedServiceAccount(
 	memberSystemNamespace *corev1.Namespace,
 	clusterName string,
 	errorOnExisting bool,
-	logger klog.Logger,
 ) (string, error) {
+	logger := klog.FromContext(ctx).WithValues("member-service-account-name", MemberServiceAccountName)
```

Comment:
```suggestion
	logger := klog.FromContext(ctx).WithValues("member-service-account-name", MemberServiceAccountName)
	ctx = klog.NewContext(ctx, logger)
```
Logger with the new key value needs to be propagated in the new context.